conans/client/tools/scm.py

```diff
@@ -146,4 +171,99 @@ def get_branch(self):
             branch = status.splitlines()[0].split("...")[0].strip("#").strip()
             return branch
         except Exception as e:
-            raise ConanException("Unable to get git branch from %s\n%s" % (self.folder, str(e)))
+            raise ConanException("Unable to get git branch from %s: %s" % (self.folder, str(e)))
+
+
+class SVN(SCMBase):
+    cmd_command = "svn"
+
+    def __init__(self, folder=None, runner=None, *args, **kwargs):
+        def runner_no_strip(command):
+            return decode_text(subprocess.check_output(command, shell=True))
+        runner = runner or runner_no_strip
+        super(SVN, self).__init__(folder=folder, runner=runner, *args, **kwargs)
+
+    def run(self, command):
+        # Ensure we always pass some params
+        return super(SVN, self).run(command="{} --no-auth-cache --non-interactive".format(command))
```

Comment:
Always "--non-interactive" could be a problem when trying to checkout a new repo (never set credentials before) in development mode (error message: "Unable to checkout ... Authentication required ..."). The common developer/user may not know something about the username / password setup method (by providing env vars) or probably this is not used in the recipe.
I suggest to add "--non-interactive" only in "conan non-interactive mode" (like "--json" or something like that) or when "trust ssl" is set to true ("--trust-server-certs" requires "--non-interactive").

Reply:
This is something interesting. Credentials are supposed to be provided to Git and SVN classes in the constructor `SCMBase::__init__(self, ...., username=None, password=None...)`, but in the workflow of the SCM feature those arguments are never used.

Different approaches:
- Using env variables (per recipe? per svn base url?) and passing them to `SVN::__init__`.
- It is up to the user to log first to the repo and have the credentials stored.
- Making it interacive (leaving the `--no-auth-cache`) so the user will be prompted for credentials each time.

I'm not sure which one is the way to go. @lasote, what do you think?

Reply:
The credentials are provided by the scm dict in recipe. There one can also use env vars. If user and password are set, you should always use them by enhancing the svn command itself.
IMHO "no-auth-cache" is only to not store credentials. But you will always be asked for when no credentials are found in existing cache.
Login first is not the best solution/workflow as the recipe could be in a different repo. Then one have to check the recipe for scm url, login and then use conan commands.